### ISSUE TITLE
My hyperealvr controller cannot move even I had setting up correct binding.

### ISSUE THREAD

**Issue by codysk** [nicoco007/DynamicOpenVR#17]
I place CustomAvatar/DynamicOpenVR at my game dir. 
But I cant move the controller in the game, they just laying on the ground.
My controller and game are work perfectly without these mods. And I have checked my controller key binding, it use the profile "Default Beat Saber Bindings".

What did I have try:
my vr_controller type name "hyperealvr_controller". It input endpoint same as vivi_controller like "/user/hand/left/pose/raw" etc.
So I create a json file in DynamicOpenVR\Bindings name "beatsaber_hypereal_controller.json" and replace the controller_type as "hyperealvr_controller". And I can see the correct binding profile "Default Beat Saber Bindings" at key binding ui.
But the controller still can not move in the game.

I check game log and find these in game log file.
```
ERROR @ 01:38:01 | IPA] DynamicOpenVR.BeatSaber: System.NullReferenceException: Object reference not set to an instance of an object
[ERROR @ 01:38:01 | IPA]   at DynamicOpenVR.BeatSaber.Plugin.OnUpdate () [0x00026] in <e74ad007f99141428267ea36f0262285>:0 
[ERROR @ 01:38:01 | IPA]   at IPA.Loader.Composite.CompositeBSPlugin+<>c.<OnUpdate>b__9_0 (IPA.Loader.PluginLoader+PluginInfo plugin) [0x00006] in <561f7537f11b496bb2465f41d54f54b0>:0 
[ERROR @ 01:38:01 | IPA]   at IPA.Loader.Composite.CompositeBSPlugin.Invoke (IPA.Loader.Composite.CompositeBSPlugin+CompositeCall callback) [0x0001d] in <561f7537f11b496bb2465f41d54f54b0>:0 
[CRITICAL @ 01:38:01 | UnityEngine] NullReferenceException: Object reference not set to an instance of an object
[CRITICAL @ 01:38:01 | UnityEngine] DynamicOpenVR.OpenVRWrapper.GetAnalogActionData (System.UInt64 actionHandle) (at <b15383f924cc4a768abc6ceece52202e>:0)
[CRITICAL @ 01:38:01 | UnityEngine] DynamicOpenVR.IO.AnalogInput.UpdateData () (at <b15383f924cc4a768abc6ceece52202e>:0)
[CRITICAL @ 01:38:01 | UnityEngine] DynamicOpenVR.OpenVRActionManager.Update () (at <b15383f924cc4a768abc6ceece52202e>:0)
[ERROR @ 01:38:01 | IPA] DynamicOpenVR.BeatSaber: System.NullReferenceException: Object reference not set to an instance of an object
[ERROR @ 01:38:01 | IPA]   at DynamicOpenVR.BeatSaber.Plugin.OnUpdate () [0x00026] in <e74ad007f99141428267ea36f0262285>:0 
[ERROR @ 01:38:01 | IPA]   at IPA.Loader.Composite.CompositeBSPlugin+<>c.<OnUpdate>b__9_0 (IPA.Loader.PluginLoader+PluginInfo plugin) [0x00006] in <561f7537f11b496bb2465f41d54f54b0>:0 
[ERROR @ 01:38:01 | IPA]   at IPA.Loader.Composite.CompositeBSPlugin.Invoke (IPA.Loader.Composite.CompositeBSPlugin+CompositeCall callback) [0x0001d] in <561f7537f11b496bb2465f41d54f54b0>:0 
[CRITICAL @ 01:38:01 | UnityEngine] NullReferenceException: Object reference not set to an instance of an object
[CRITICAL @ 01:38:01 | UnityEngine] DynamicOpenVR.OpenVRWrapper.GetAnalogActionData (System.UInt64 actionHandle) (at <b15383f924cc4a768abc6ceece52202e>:0)
[CRITICAL @ 01:38:01 | UnityEngine] DynamicOpenVR.IO.AnalogInput.UpdateData () (at <b15383f924cc4a768abc6ceece52202e>:0)
[CRITICAL @ 01:38:01 | UnityEngine] DynamicOpenVR.OpenVRActionManager.Update () (at <b15383f924cc4a768abc6ceece52202e>:0)
[ERROR @ 01:38:01 | IPA] DynamicOpenVR.BeatSaber: System.NullReferenceException: Object reference not set to an instance of an object
[ERROR @ 01:38:01 | IPA]   at DynamicOpenVR.BeatSaber.Plugin.OnUpdate () [0x00026] in <e74ad007f99141428267ea36f0262285>:0 
[ERROR @ 01:38:01 | IPA]   at IPA.Loader.Composite.CompositeBSPlugin+<>c.<OnUpdate>b__9_0 (IPA.Loader.PluginLoader+PluginInfo plugin) [0x00006] in <561f7537f11b496bb2465f41d54f54b0>:0 
[ERROR @ 01:38:01 | IPA]   at IPA.Loader.Composite.CompositeBSPlugin.Invoke (IPA.Loader.Composite.CompositeBSPlugin+CompositeCall callback) [0x0001d] in <561f7537f11b496bb2465f41d54f54b0>:0 
[CRITICAL @ 01:38:01 | UnityEngine] NullReferenceException: Object reference not set to an instance of an object
[CRITICAL @ 01:38:01 | UnityEngine] DynamicOpenVR.OpenVRWrapper.GetAnalogActionData (System.UInt64 actionHandle) (at <b15383f924cc4a768abc6ceece52202e>:0)
[CRITICAL @ 01:38:01 | UnityEngine] DynamicOpenVR.IO.AnalogInput.UpdateData () (at <b15383f924cc4a768abc6ceece52202e>:0)
[CRITICAL @ 01:38:01 | UnityEngine] DynamicOpenVR.OpenVRActionManager.Update () (at <b15383f924cc4a768abc6ceece52202e>:0)
```

**Comment by nicoco007** [comment]
Could you try using [this version](https://ci.gnyra.com/job/DynamicOpenVR/job/master/108/artifact/DynamicOpenVR.BeatSaber-a1d901d-RELEASE.zip) of DynamicOpenVR? I'm not sure it'll fix your problem but it should hopefully give some more info on what's going on.

**Comment by codysk** [comment]
I'm not sure that was same problem or not.
The game was totally crash at launching after I use version a1d901d .

Seems like openvr interface initialize failed.
```
[INFO @ 16:12:58 | Versus] No ignored songs.
[INFO @ 16:12:58 | DynamicOpenVR.BeatSaber] Starting DynamicOpenVR.BeatSaber
[DEBUG @ 16:12:58 | IPA] Compressing log file E:\Steam\steamapps\common\Beat Saber\Logs\DynamicOpenVR.BeatSaber\2020.02.17.15.53.11.log
[ERROR @ 16:12:58 | DynamicOpenVR.BeatSaber] Failed to initialize DynamicOpenVR.BeatSaber.
[ERROR @ 16:12:58 | DynamicOpenVR.BeatSaber] DynamicOpenVR.OpenVRInitException: Failed to initialize OpenVR: Init_InterfaceNotFound
[ERROR @ 16:12:58 | DynamicOpenVR.BeatSaber]   at DynamicOpenVR.OpenVRStatus.Init () [0x00018] in <a2f6e15e5ca745d79b385da0e02e3cc5>:0 
[ERROR @ 16:12:58 | DynamicOpenVR.BeatSaber]   at DynamicOpenVR.BeatSaber.Plugin.OnApplicationStart () [0x00045] in <9a862b9daa3e4bb68d6f71c29a3d0c93>:0 
```

**Comment by codysk** [comment]
I know what happened.
I use Steam VR 1.3 because controller driver cant work at newer version.
And DynamicOpenVR use an api call `GetPoseActionDataForNextFrame` which only can use on openvr 1.4 or newer version. So it cause an exception at get controller pose.

I solved this problem at [this commit](https://github.com/codysk/DynamicOpenVR/commit/4745f8fa1852fff23a90d8c8fb8a22fda0964532). Avatar animation also works, everything is ok.
But the solution is downgrade the openvr sdk and use old method.So I'm not sure if should open a pull request for it.

**Comment by nicoco007** [comment]
You can definitely go ahead and open a pull request! I'll just have to check to make sure it doesn't break anything with the latest version of SteamVR.

**Comment by nicoco007** [comment]
Changes were merged in f3a422d. Thanks for your help!
